tests/js/core/util/path/path.js

```diff
@@ -1,179 +1,204 @@
-		equal(path.isEmbedded("#page"), true, "isEmbedded for \"#page\"");
-		equal(path.isEmbedded("http://localhost"), false, "isEmbedded for \"http://localhost\"");
-		equal(path.isEmbedded("/index.html?a=1#page"), true, "isEmbedded for \"/index.html?s=query#page\"");
-	});
-
-	test("squash", function () {
-		equal(path.squash("/path", "http://localhost/something/index.html#page"), "http://localhost/path", "squash for \"/path\", \"http://localhost/something/index.html#page\"");
-		equal(path.squash("#something", "http://localhost/path/index.html#page?search"), "http://localhost/path/index.html#something", "squash for \"#something\" \"http://localhost/path/index.html#page?search\"");
-	});
-
-	test("isPreservableHash", function () {
-		equal(path.isPreservableHash("#&ui-state"), true, "isPreservableHash for \"#&ui-state\"");
-		equal(path.isPreservableHash("#start"), false, "#start");
-	});
-
-	test("hashToSelector", function () {
-		equal(path.hashToSelector("something"), "something", "hashToSelector for \"something\"");
-		equal(path.hashToSelector("#something"), "#something", "hashToSelector for \"#something\"");
-		equal(path.hashToSelector("#som?e:t{h}i|ng"), "#som\\?e\\:t\\{h\\}i\\|ng", "hashToSelector for \"#som?e:t{h}i|ng\"");
-	});
-
-	test("getFilePath", function () {
-		equal(path.getFilePath("/path/index.html&http://localhost/path/something#page"), "/path/index.html", "Return the file path");
-	});
-
-	test("getDocumentBase", function () {
-		equal(typeof path.getDocumentBase(false), "string", "Get document base for asParsedObject as false");
-		equal(typeof path.getDocumentBase(true), "object", "Get document base for asParsedObject as true");
-	});
-
-	test("getClosestBaseUrl", function () {
-		equal(path.getClosestBaseUrl(document.getElementById("test1"), pageSelector),
-			window.location.origin + window.location.pathname,
-			"Return the substring of a filepath before the sub-page key, for making a server request"
-		);
-	});
-
-	test("getLocation", function () {
-		var popupUrl = "http://localhost/something/index.html#?popup=true";
-
-		equal(path.getLocation(popupUrl), popupUrl, "Popup url.");
-		popupUrl = "http://jblas:password@mycompany.com:8080/mail/inbox?msg=1234&type=unread#msg-content?param1=true&param2=123";
-		equal(path.getLocation(popupUrl), "http://mycompany.com:8080/mail/inbox?msg=1234&type=unread#msg-content?param1=true&param2=123", "Abstract url.");
-	});
-});
+	function runTests(path, helpers, tau) {
+
+		var pageSelector = ".ui-page";
+
+		tau = tau || window.ns;
+		path = path || window.ns.util.path;
+		helpers = helpers || window.helpers;
+
+		function initHTML() {
+			var HTML = helpers.loadHTMLFromFile("/base/tests/js/core/util/path/test-data/sample.html"),
+				parent = document.getElementById("qunit-fixture") || helpers.initFixture();
+
+			parent.innerHTML = HTML;
+		}
+
+		module("core/util/path", {
+			setup: initHTML
+		});
+
+		test("getDocumentUrl", function () {
+			var urlObject = {
+
```

Comment:
delete line. and attach below line to upper line. I think merge line that looks good.

Reply:
done